sdk/python/kfp/compiler/compiler.py

```diff
@@ -128,14 +128,16 @@ def _op_to_template(self, op):
       template['container']['command'] = op.command
 
     # Set resources.
-    if op.memory_limit or op.cpu_limit or op.memory_request or op.cpu_request:
+    if op.memory_limit or op.cpu_limit or op.memory_request or op.cpu_request or op.nvidia_gpu_limit:
       template['container']['resources'] = {}
-    if op.memory_limit or op.cpu_limit:
+    if op.memory_limit or op.cpu_limit or op.nvidia_gpu_limit:
```

Comment:
For each resource there is a "limit" and a "request". Do you also have the change for set_gpu_request()?

Reply:
I have tried requests/gpu which doesn't work as expected. According to https://cloud.google.com/kubernetes-engine/docs/how-to/gpus, it only mentions the limit support. It looks like not supported.

Reply:
Found this: https://kubernetes.io/docs/tasks/manage-gpus/scheduling-gpus/

"GPUs are only supposed to be specified in the limits section, which means:
You can specify GPU limits without specifying requests because Kubernetes will use the limit as the request value by default.
You can specify GPU in both limits and requests but these two values must be equal.
You cannot specify GPU requests without specifying limits."

The naming in the doc is misleading to me. limits sound like max, and requests min but actually it means both max and min here. This is different from cpu and memory.

Is it a good idea to stop the confusion at DSL side, say we just call it set_gpu()?

Reply:
I feel it's better to just leave it open and let user to specify the values by add_resource_limit or add_resource_request. It will be easier for user to follow the instructions from gke document and less maintenance for us if things are changed in the future, e.g. gke start supports other brand gpus.

Reply:
SG. Add the link to docstring then?

Reply:
Ah. You removed the explicit gpu API. Then it is fine.

Reply:
hmm... maybe it is also worthwhile to keep the set_gpu_limit()? a lot of python users (notebook users) they rely on intellisense. Having an explicit GPU setter helps them find it easily. Also I would prefer set_gpu* so auto complete can help here. What do you think of set_gpu_nvidia_limit()? Other suggestions? With only set_resource_limit and both cpu/memory setter, it is harder and confusing to users as to whether/how to set GPU.

Reply:
Done